larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewSimpleTracksTool.cc

```diff
@@ -0,0 +1,127 @@
+/**
+ *  @file   larpandoracontent/LArThreeDReco/LArShowerMatching/TwoViewSimpleTracksTool.cc
+ *
+ *  @brief  Implementation of the clear showers tool class.
+ *
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+
+#include "larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewSimpleTracksTool.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+TwoViewSimpleTracksTool::TwoViewSimpleTracksTool() :
+    m_minMatchedFraction(0.2f),
+    m_minMatchingScore(0.9f),
+    m_minMatchedSamplingPoints(5),
+    m_minXOverlapFraction(0.5f)
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool TwoViewSimpleTracksTool::Run(TwoViewTransverseTracksAlgorithm *const pAlgorithm, MatrixType &overlapMatrix)
+{
+    if (PandoraContentApi::GetSettings(*pAlgorithm)->ShouldDisplayAlgorithmInfo())
+       std::cout << "----> Running Algorithm Tool: " << this->GetInstanceName() << ", " << this->GetType() << std::endl;
+
+    ProtoParticleVector protoParticleVector;
+    this->FindBestTrack(overlapMatrix, protoParticleVector);
+
+    const bool particlesMade(pAlgorithm->CreateThreeDParticles(protoParticleVector));
+    return particlesMade;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void TwoViewSimpleTracksTool::FindBestTrack(const MatrixType &overlapMatrix, ProtoParticleVector &protoParticleVector) const
+{
+    ClusterVector sortedKeyClusters;
+    overlapMatrix.GetSortedKeyClusters(sortedKeyClusters);
+
+    for (const Cluster *const pKeyCluster : sortedKeyClusters)
+    {
+        if (!pKeyCluster->IsAvailable())
+            continue;
+
+        unsigned int n0(0), n1(0);
+        MatrixType::ElementList elementList;
+        overlapMatrix.GetConnectedElements(pKeyCluster, true, elementList, n0, n1);
+
+        if (elementList.empty())
+            continue;
+
+        MatrixType::Element bestElement(elementList.back());
+
+        for (MatrixType::ElementList::const_reverse_iterator iIter = elementList.rbegin(), iIterEnd = elementList.rend(); iIter != iIterEnd; ++iIter)
```

Comment:
(Here's where you're relying on consistency with the OverlapResult `operator<` defined before.)

Reply:
@PandoraPFA, did you have a specific suggestion regarding these lines? Thanks!

Reply:
No, was rather just highlighting that this ordered depends exactly on the ordering you defined in the OverlapResult `operator<`, so any changes there would have an impact here. (Strictly, you're sorting the OverlapElements, but these are just OverlapResults, with the associated clusters, and redirect the sorting to the OverlapResult). So, looks OK to me, with the operator putting the "worst" elements first, then you using a reverse iterator to see the "best" elements first.

Reply:
Okay, thank you! :)